docs/otlp/getting-started-aspnetcore/README.md

```diff
@@ -0,0 +1,187 @@
+    .UseOtlpExporter();
+```
+
+The `AddOpenTelemetry` extension is part of the
+[OpenTelemetry.Extensions.Hosting](../../../src/OpenTelemetry.Extensions.Hosting/README.md)
+package and registers the OpenTelemetry SDK into the host. For more details see:
+[Initialize the SDK using a host](../../README.md#initialize-the-sdk-using-a-host).
+
+The `tracing.AddAspNetCoreInstrumentation()` and
+`metrics.AddAspNetCoreInstrumentation()` calls register AspNetCore
+instrumentation. For more details see:
+[OpenTelemetry.Instrumentation.AspNetCore](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Instrumentation.AspNetCore/README.md).
+
+The `UseOtlpExporter` extension configures the OpenTelemetry .NET OTLP exporter
+for logging, metrics, and tracing. For more details see: [Enable OTLP Exporter
+for all
+signals](../../../src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md#enable-otlp-exporter-for-all-signals).
+
+The programs maps the index route ("/") to write out the OpenTelemetry trace
+information on the response:
+
+```csharp
+app.MapGet("/", () => $"Hello World! OpenTelemetry Trace: {Activity.Current?.Id}");
+```
+
+In OpenTelemetry .NET the [Activity
+class](https://learn.microsoft.com/dotnet/api/system.diagnostics.activity)
+represents the OpenTelemetry Specification
+[Span](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#span).
+For more details about how the OpenTelemetry Specification is implemented in
+.NET see: [Introduction to OpenTelemetry .NET Tracing
+API](../../../src/OpenTelemetry.Api/README.md#introduction-to-opentelemetry-net-tracing-api).
+
+## Next steps
+
+Explore and add relevant [instrumentation
+libraries](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/glossary.md#instrumentation-library)
+and resource detectors to your OpenTelemetry configuration by visiting the
+[opentelemetry-dotnet-contrib
+repository](https://github.com/open-telemetry/opentelemetry-dotnet-contrib)
+and/or the [OpenTelemetry
+registry](https://opentelemetry.io/ecosystem/registry/?language=dotnet).
+Instrumentation libraries help automatically generate telemetry for common
+application tasks such as making an [outgoing HTTP
+call](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Instrumentation.Http).
+Resource detectors help decorate telemetry with information about the hosting
+environment.
+
+Use the
+[ActivitySource](https://learn.microsoft.com/dotnet/api/system.diagnostics.activitysource),
+[Meter](https://learn.microsoft.com/dotnet/api/system.diagnostics.metrics.meter),
+and
+[ILogger&lt;T&gt;](https://learn.microsoft.com/dotnet/api/microsoft.extensions.logging.ilogger-1)
+APIs to add custom telemetry to your application and/or libraries. The [Getting
+Started with OpenTelemetry Protocol (OTLP) in 5 Minutes - Console
```

Comment:
This (custom telemetry manually) is something that needs some thoughts around how to organize. This is currently only shown in the [example](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/examples/AspNetCore/Controllers/WeatherForecastController.cs#L45-L64) app only, and we have seen many users making mistakes (like creating Instruments repeatedly, ActivitySource repeatedily etc.)

Reply:
We should rationalize how much info should be here in this repo, and how much is in the .NET documentation. The latter info about how to create telemetry is not specific to OTel so could/should probably go into the respective sections of the .NET docs, which can be linked to from here

Reply:
Yes agreed.
Custom telemetry itself is a lot of content!
1. How to create a new Activity 
2. How to add an extra Tag or Event to existing Activity. (Activity.Current mostly)
3. How to do 2 in a generic way - (ActivityProcessor).
4. How to do 2 in instrumentation specific way - Enrich capabilities offered by specific instrumentation libraries.
5. Advice on diff. between ActivityEvent and ILogger.

Reply:
Either of you have some doc in mind you want me to link to instead of what I have at the moment or can this be a follow-up effort?

Reply:
follow up only. I dont have a great answer at the moment.